From 0a6457836918f0aba88f861b7be65bfb883e6780 Mon Sep 17 00:00:00 2001
From: Holger Freyther <ich@tamarin.(none)>
Date: Tue, 13 May 2008 18:53:40 +0100
Subject: [PATCH] [pcf50633] Print less debugging messages
     According to the comments in the code we have to read
     INT1 - INT5 in case of any interrupt. So even if we have masked
     out the periodic timer interrupt we were able to see that it
     fired. Only print that we got a SECOND interrupt if we are actually
     interested in the periodic tick. From a brief look at the datasheet there
     is no way to stop the periodic tick.

Signed-Off-By: Holger Freyther <zecke@openmoko.org>
---
 drivers/i2c/chips/pcf50633.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/i2c/chips/pcf50633.c b/drivers/i2c/chips/pcf50633.c
index 1c10c42..b354a9c 100644
--- a/drivers/i2c/chips/pcf50633.c
+++ b/drivers/i2c/chips/pcf50633.c
@@ -686,9 +686,10 @@ static void pcf50633_work(struct work_struct *work)
 			rtc_update_irq(pcf->rtc, 1, RTC_AF | RTC_IRQF);
 	}
 	if (pcfirq[0] & PCF50633_INT1_SECOND) {
-		DEBUGPC("SECOND ");
-		if (pcf->flags & PCF50633_F_RTC_SECOND)
+		if (pcf->flags & PCF50633_F_RTC_SECOND) {
+			DEBUGPC("SECOND ");
 			rtc_update_irq(pcf->rtc, 1, RTC_PF | RTC_IRQF);
+		}
 
 		if (pcf->onkey_seconds >= 0 &&
 		    pcf->flags & PCF50633_F_PWR_PRESSED) {
-- 
1.5.6.5

